nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/CoreFeature.java

```diff
@@ -92,6 +93,40 @@
 @LocalDeploy("org.nuxeo.ecm.core.event:test-queuing.xml")
 public class CoreFeature extends SimpleFeature {
 
+    public class WorksWaiter implements Waiter {
+        @Override
+        public boolean await(long deadline) throws InterruptedException {
+            final WorkManager mgr = Framework.getService(WorkManager.class);
+            if (mgr.awaitCompletion(deadline - System.currentTimeMillis(), TimeUnit.MILLISECONDS)) {
+                return true;
+            }
+            logInfos(mgr);
+            return false;
+        }
+
+        protected void logInfos(final WorkManager mgr) {
+            String message = "Timed out while waiting for works";
+            StringBuffer bf = new StringBuffer();
+            bf.append(message + " ");
+            Iterator<String> ids = mgr.getWorkQueueIds().iterator();
```

Comment:
No need for an interator / while, you can do a foreach loop on `mgr.getWorkQueueIds()` directly.